cmake/Dependencies.cmake

```diff
@@ -1324,6 +1324,55 @@ if(USE_ROCM AND ROCM_VERSION_DEV VERSION_LESS "5.2.0")
   include_directories(SYSTEM ${THRUST_PATH})
 endif()
 
+# ---[ XPU
+if(USE_XPU)
+  include(${CMAKE_CURRENT_LIST_DIR}/public/xpu.cmake)
+  # Support GCC only in the moment.
+  if(PYTORCH_FOUND_XPU AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # # -- Host flags (SYCL_CXX_FLAGS)
+    # Combine with torch_compile_options
+    list(APPEND SYCL_HOST_FLAGS -fPIC)
+    list(APPEND SYCL_HOST_FLAGS -std=c++17)
+    # SYCL headers warnings
+    list(APPEND SYCL_HOST_FLAGS -Wno-deprecated-declarations)
+    list(APPEND SYCL_HOST_FLAGS -Wno-attributes)
+
+    if(CMAKE_BUILD_TYPE MATCHES Debug)
+       list(APPEND SYCL_HOST_FLAGS -g)
+       list(APPEND SYCL_HOST_FLAGS -O0)
+    endif(CMAKE_BUILD_TYPE MATCHES Debug)
+
+    # -- Kernel flags (SYCL_KERNEL_OPTIONS)
+    # The fast-math will be enabled by default in SYCL compiler.
+    # Refer to [https://clang.llvm.org/docs/UsersManual.html#cmdoption-fno-fast-math]
+    # 1. We enable below flags here to be warn about NaN and Infinity,
+    # which will be hidden by fast-math by default.
+    # 2. The associative-math in fast-math allows floating point
+    # operations to be reassociated, which will lead to non-deterministic results.
```

Comment:
can you explain why it leads to non-deterministic results?

Reply:
It's not clear. I will change it. Should be non-deterministic to intention of programmer.